extensions/wikia/AdEngine3/src/targeting.js

```diff
@@ -165,6 +165,21 @@ function getZone(adsContext) {
 	};
 }
 
+function getLikhoParams() {
+  const likhoStorage = JSON.parse(localStorage.getItem('likho')) || [];
+  const params = [];
+
+  likhoStorage.forEach((item, index, object) => {
+    if (item.expirationTime > Date.now()) {
+      params.push(item.likhoType);
+    } else {
+      object.splice(index, 1);
```

Comment:
I am not sure about that. It modifies array on which it performs actions. I am not saying no explicitly but I am not sure what will happen .